### PR TITLE
Additional Dependencies

### PR DESCRIPTION
**Pull request by sahiljhawar**
@atoivonen13  Reqs and test for additional dependencies. Move #369 here

Installation CI only runs when *_requirements.txt or toml file is changed

### CODE REVIEW

File: .github/workflows/dependency-installation.yml
Comment (sahiljhawar) [inline .github/workflows/dependency-installation.yml:50]: change git link after merge

**Comment by sahiljhawar** [comment]
@atoivonen13 you can add more files here

**Comment by atoivonen13** [comment]
@sahiljhawar you want me to move the changes from https://github.com/nuclear-multimessenger-astronomy/nmma/pull/369 here?

**Comment by sahiljhawar** [comment]
@sahiljhawar This PR already contains the changes you made in #369. I mean if you have more optional dependency files, you can add them in this PR.

**Comment by sahiljhawar** [comment]
@atoivonen13 This PR already contains the changes you made in https://github.com/nuclear-multimessenger-astronomy/nmma/pull/369. I mean, if you have more optional dependency files, you can add them in this PR.

**Comment by atoivonen13** [comment]
@sahiljhawar it does not appear I can push to your fork, but you can add sklearn as its own file and optional dependency. I think the thought was if we were going to make tensorflow optional, we might as well make sklearn option, as users usually use one or the other

**Comment by sahiljhawar** [comment]
okay. But there should be atleast one ML package which installs with NMMA and anyone can right away start with very basic analysis. What do you think @atoivonen13 ?

**Comment by atoivonen13** [comment]
@sahiljhawar first off, looks like pyFFTW made a release. Second I was just chatting with Michael about making sklearn a separate file and he thought it makes sense to be agnostic to the methods and just be clear in documentation

**Comment by sahiljhawar** [comment]
@atoivonen13 yes I know about PyFFTW, already pushed Python 3.12 updates to the repo. also added new file for sklearn

**Comment by sahiljhawar** [comment]
I guess, need to write a few notes on MacOS installation (and Python 3.12 in particular)

**Comment by atoivonen13** [comment]
Thanks @sahiljhawar I was mainly just checking in to see when this would be merged and we would do a release. Is there anything else I need to do here?

**Comment by sahiljhawar** [comment]
@atoivonen13, I think this is done. Waiting for @mcoughlin approval for merge.
Also, do we need make a release right now (0.2.1)?

**Review by mcoughlin — APPROVED** [review]
Sorry! LGTM

**Comment by atoivonen13** [comment]
@sahiljhawar A release would be nice when you can so I can make sure everything works with em-bright, but it does not need to be done today or anything. I figured between this and the pyfftw a release is merited, are we waiting on anything else to include in 0.2.1?

**Comment by sahiljhawar** [comment]
I remember from one of the NMMA calls that Henrik has something to include in the next release, however I am not super sure about it.

**Comment by sahiljhawar** [comment]
@atoivonen13 @mcoughlin Done!

**Pull request merged**
